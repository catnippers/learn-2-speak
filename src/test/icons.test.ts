import {describe, expect, test} from "vitest";
import {mount} from "@vue/test-utils";

import GitHubLink from "../app/components/icons/GitHubLink.vue";
import RefreshIcon from "../app/components/icons/RefreshIcon.vue";
import SpeakerIcon from "../app/components/icons/SpeakerIcon.vue";
import SpeakerOffIcon from "../app/components/icons/SpeakerOffIcon.vue";

describe('mount icons', () => {
    test('mount GitHubLink component', async () => {
        expect(GitHubLink).toBeTruthy();

        const wrapper = mount(GitHubLink);

        expect(wrapper.exists());
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount RefreshIcon component', async () => {
        expect(RefreshIcon).toBeTruthy();

        const wrapper = mount(RefreshIcon);

        expect(wrapper.exists());
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount SpeakerIcon component', async () => {
        expect(SpeakerIcon).toBeTruthy();

        const wrapper = mount(SpeakerIcon);

        expect(wrapper.exists());
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount SpeakerOffIcon component', async () => {
        expect(SpeakerOffIcon).toBeTruthy();

        const wrapper = mount(SpeakerOffIcon);

        expect(wrapper.exists());
        expect(wrapper.html()).toMatchSnapshot();
    });
})
