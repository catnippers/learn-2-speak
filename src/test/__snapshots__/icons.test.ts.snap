// Vitest Snapshot v1

exports[`mount icons > mount GitHubLink component 1`] = `
"<a href=\\"https://github.com/catnippers/learn-2-speak/\\" target=\\"_blank\\" class=\\"lts-github-icon-corner\\" data-test=\\"lts-github-icon-corner\\"><svg viewBox=\\"0 0 250 250\\" class=\\"lts-github-icon\\">
    <path d=\\"M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z\\"></path>
    <path d=\\"M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2\\" fill=\\"currentColor\\"></path>
    <path d=\\"M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z\\" fill=\\"currentColor\\"></path>
  </svg></a>"
`;

exports[`mount icons > mount RefreshIcon component 1`] = `
"<div class=\\"lts-refresh-icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 15 15\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M1.90321 7.29677C1.90321 10.341 4.11041 12.4147 6.58893 12.8439C6.87255 12.893 7.06266 13.1627 7.01355 13.4464C6.96444 13.73 6.69471 13.9201 6.41109 13.871C3.49942 13.3668 0.86084 10.9127 0.86084 7.29677C0.860839 5.76009 1.55996 4.55245 2.37639 3.63377C2.96124 2.97568 3.63034 2.44135 4.16846 2.03202L2.53205 2.03202C2.25591 2.03202 2.03205 1.80816 2.03205 1.53202C2.03205 1.25588 2.25591 1.03202 2.53205 1.03202L5.53205 1.03202C5.80819 1.03202 6.03205 1.25588 6.03205 1.53202L6.03205 4.53202C6.03205 4.80816 5.80819 5.03202 5.53205 5.03202C5.25591 5.03202 5.03205 4.80816 5.03205 4.53202L5.03205 2.68645L5.03054 2.68759L5.03045 2.68766L5.03044 2.68767L5.03043 2.68767C4.45896 3.11868 3.76059 3.64538 3.15554 4.3262C2.44102 5.13021 1.90321 6.10154 1.90321 7.29677ZM13.0109 7.70321C13.0109 4.69115 10.8505 2.6296 8.40384 2.17029C8.12093 2.11718 7.93465 1.84479 7.98776 1.56188C8.04087 1.27898 8.31326 1.0927 8.59616 1.14581C11.4704 1.68541 14.0532 4.12605 14.0532 7.70321C14.0532 9.23988 13.3541 10.4475 12.5377 11.3662C11.9528 12.0243 11.2837 12.5586 10.7456 12.968L12.3821 12.968C12.6582 12.968 12.8821 13.1918 12.8821 13.468C12.8821 13.7441 12.6582 13.968 12.3821 13.968L9.38205 13.968C9.10591 13.968 8.88205 13.7441 8.88205 13.468L8.88205 10.468C8.88205 10.1918 9.10591 9.96796 9.38205 9.96796C9.65819 9.96796 9.88205 10.1918 9.88205 10.468L9.88205 12.3135L9.88362 12.3123C10.4551 11.8813 11.1535 11.3546 11.7585 10.6738C12.4731 9.86976 13.0109 8.89844 13.0109 7.70321Z\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\"></path>
  </svg></div>"
`;

exports[`mount icons > mount SpeakerIcon component 1`] = `
"<div class=\\"lts-speaker-icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 15 15\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M7.46968 1.05085C7.64122 1.13475 7.75 1.30904 7.75 1.5V13.5C7.75 13.691 7.64122 13.8653 7.46968 13.9492C7.29813 14.0331 7.09377 14.0119 6.94303 13.8947L3.2213 11H1.5C0.671571 11 0 10.3284 0 9.5V5.5C0 4.67158 0.671573 4 1.5 4H3.2213L6.94303 1.10533C7.09377 0.988085 7.29813 0.966945 7.46968 1.05085ZM6.75 2.52232L3.69983 4.89468C3.61206 4.96294 3.50405 5 3.39286 5H1.5C1.22386 5 1 5.22386 1 5.5V9.5C1 9.77615 1.22386 10 1.5 10H3.39286C3.50405 10 3.61206 10.0371 3.69983 10.1053L6.75 12.4777V2.52232ZM10.2784 3.84804C10.4623 3.72567 10.7106 3.77557 10.833 3.95949C12.2558 6.09798 12.2558 8.90199 10.833 11.0405C10.7106 11.2244 10.4623 11.2743 10.2784 11.1519C10.0944 11.0296 10.0445 10.7813 10.1669 10.5973C11.4111 8.72728 11.4111 6.27269 10.1669 4.40264C10.0445 4.21871 10.0944 3.97041 10.2784 3.84804ZM12.6785 1.43044C12.5356 1.2619 12.2832 1.24104 12.1147 1.38386C11.9462 1.52667 11.9253 1.77908 12.0681 1.94762C14.7773 5.14488 14.7773 9.85513 12.0681 13.0524C11.9253 13.2209 11.9462 13.4733 12.1147 13.6161C12.2832 13.759 12.5356 13.7381 12.6785 13.5696C15.6406 10.0739 15.6406 4.92612 12.6785 1.43044Z\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\"></path>
  </svg></div>"
`;

exports[`mount icons > mount SpeakerOffIcon component 1`] = `
"<div class=\\"lts-speaker-icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 15 15\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M7.72361 1.05279C7.893 1.13749 8 1.31062 8 1.5V13.5C8 13.6894 7.893 13.8625 7.72361 13.9472C7.55421 14.0319 7.35151 14.0136 7.2 13.9L3.33333 11H1.5C0.671573 11 0 10.3284 0 9.5V5.5C0 4.67158 0.671573 4 1.5 4H3.33333L7.2 1.1C7.35151 0.986371 7.55421 0.968093 7.72361 1.05279ZM7 2.5L3.8 4.9C3.71345 4.96491 3.60819 5 3.5 5H1.5C1.22386 5 1 5.22386 1 5.5V9.5C1 9.77614 1.22386 10 1.5 10H3.5C3.60819 10 3.71345 10.0351 3.8 10.1L7 12.5V2.5ZM14.8536 5.14645C15.0488 5.34171 15.0488 5.65829 14.8536 5.85355L13.2071 7.5L14.8536 9.14645C15.0488 9.34171 15.0488 9.65829 14.8536 9.85355C14.6583 10.0488 14.3417 10.0488 14.1464 9.85355L12.5 8.20711L10.8536 9.85355C10.6583 10.0488 10.3417 10.0488 10.1464 9.85355C9.95118 9.65829 9.95118 9.34171 10.1464 9.14645L11.7929 7.5L10.1464 5.85355C9.95118 5.65829 9.95118 5.34171 10.1464 5.14645C10.3417 4.95118 10.6583 4.95118 10.8536 5.14645L12.5 6.79289L14.1464 5.14645C14.3417 4.95118 14.6583 4.95118 14.8536 5.14645Z\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\"></path>
  </svg></div>"
`;
